src/web/status.rs

```diff
@@ -0,0 +1,143 @@
+use super::cache::CachePolicy;
+use crate::{
+    db::Pool,
+    utils::spawn_blocking,
+    web::{error::AxumResult, match_version_axum},
+};
+use axum::{
+    extract::{Extension, Path},
+    http::header::ACCESS_CONTROL_ALLOW_ORIGIN,
+    response::IntoResponse,
+    Json,
+};
+
+pub(crate) async fn status_handler(
+    Path((name, req_version)): Path<(String, String)>,
+    Extension(pool): Extension<Pool>,
+) -> AxumResult<impl IntoResponse> {
+    let (_, id) = match_version_axum(&pool, &name, Some(&req_version))
+        .await?
+        .assume_exact()?
+        .assume_exact()?;
```

Comment:
nit: should we adapt the names of both methods to avoid confusion?

Reply:
Actually, I just realized that shields.io uses `/latest`, so we need to support that, not sure why I thought it was using versioned urls.